pkg/apis/serving/v1beta1/component.go

```diff
@@ -106,6 +107,10 @@ type ComponentExtensionSpec struct {
 	// More info: http://kubernetes.io/docs/user-guide/annotations
 	// +optional
 	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// RawDeploymentSpec customizes the deployment configuration for the predictor when using RawDeployment mode.
+	// +optional
+	RawDeploymentSpec *appsv1.DeploymentSpec `json:"rawDeploymentSpec,omitempty"`
```

Comment:
Let's discuss the design, there are conflicting fields with Serverless mode(Knative)..

Reply:
Also this includes the entire podTemplate field which is dup of PodSpec on the component